### PR TITLE
remove confusing extra information

### PR DESCRIPTION
**Pull request by chaitanyaj2222**
### Description
It conveys the information that only multi core processors can help with multi threading which as far as I know is incorrect. The concept of multithreading was there even in single core processors. 

However, the advantage that multicore processors offer is that it can allow for a thread to be executed in in another core when any core of the multicore processor is free.

### Motivation
I believe that this change could bring in more clarity to the topic

### Related issues and pull requests
Relates to #28719

### CODE REVIEW

**Comment by wbamberg** [comment]
Thanks for your PR! But I don't think that sentence suggests that you can't have multithreaded code on a single-core processor. It says that if you are running on a multi-core processor, then you can take advantage of that if your code is multithreaded.

**Comment by 7777chaitanya** [comment]
Since it is explicitly mentioned that we can take advantage of modern multicore processors, I immediately understood it as only Multicore processors will be able to execute multiple threads. That's why I raised this PR.

File: files/en-us/glossary/thread/index.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Thread in computer science is the execution of running multiple tasks or program
...
+However, modern {{Glossary("JavaScript")}} offers ways to create additional threads, each executing independently while possibly communicating between one another. This is done using technologies such as **[web workers](/en-US/docs/Web/API/Web_Workers_API)**, which can be used to spin off a sub-program which runs concurrently with the main thread in a thread of its own. This allows slow, complex, or long-running tasks to be executed independently of the main thread, preserving the overall performance of the site or app—as well as that of the browser overall.
```
Comment (wbamberg) [inline files/en-us/glossary/thread/index.md:13]:
```suggestion
However, modern {{Glossary("JavaScript")}} offers ways to create additional threads, each executing independently while possibly communicating between one another. This is done using technologies such as **[web workers](/en-US/docs/Web/API/Web_Workers_API)**, which can be used to spin off a sub-program which runs concurrently with the main thread in a thread of its own. This allows slow, complex, or long-running tasks to be executed independently of the main thread, preserving the overall performance of the site or app—as well as that of the browser overall.

Single-threaded and multi-threaded programs can both run on either single-core or multi-core processors. A single-threaded program will only use a single core, even on a multi-core processor. However, a multi-threaded program can use multiple cores when running on a multi-core processor, which gives it a performance boost over a single-threaded program.
```

Maybe? This seems like a lot of explanation.

**Comment by wbamberg** [comment]
> I immediately understood it as only Multicore processors will be able to execute multiple threads.

It doesn't say that though. It says multithreaded code can "take advantage of" multicore processors, meaning that it can use multiple cores, while singlethreaded code can't.

Just deleting that sentence loses something useful from the page. I've added a suggestion, but not sure it is an improvement over the original really. @hamishwillee ?

File: files/en-us/glossary/thread/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/glossary/thread/index.md:13]: @wbamberg The whole page needs to be re-written - the first sentence and last paragraph to me are just wrong.

I agree with you that the removed text is not ambiguous. It is poor though - what's the "this"? 
If it is worth talking about then having more words is OK. But my focus would be multi threaded on single vs mult-core.

How about this compromise - not so many words as yours and cleans up the specific misperception. 

```suggestion
Modern {{Glossary("JavaScript")}} offers ways to create additional threads, each executing independently while possibly communicating between one another. This is done using technologies such as [web workers](/en-US/docs/Web/API/Web_Workers_API), which can be used to spin off a sub-program that runs concurrently with the main thread in a thread of its own. This allows slow, complex, or long-running tasks to be executed independently of the main thread, preserving the overall performance of the site or app—as well as that of the browser overall.
Threading also allows web applications to take advantage of modern multi-core processors: allowing for even better performance than a multi-threaded application running on a single core.
```

Comment (wbamberg) [reply files/en-us/glossary/thread/index.md:13]: > @wbamberg The whole page needs to be re-written - the first sentence and last paragraph to me are just wrong.

Yes, I agree, this is not a good article. But this PR is not an improvement.

> 
> I agree with you that the removed text is not ambiguous. It is poor though - what's the "this"? If it is worth talking about then having more words is OK. But my focus would be multi threaded on single vs mult-core.
> 
> How about this compromise - not so many words as yours and cleans up the specific misperception.

I mean, it's fine, and an improvement over what's there, but this:

> enabling even better performance than multi-threaded applications running on a single core.

For me the main point here is that single-threaded programs can't take advantage of multicore, and multi-threaded programs can.

But maybe just merge this, and file another PR to try to rewrite the page?

Comment (hamishwillee) [reply files/en-us/glossary/thread/index.md:13]: Sure! We'll be no worse off. Are you happy to take a shot at the rewrite - I suspect you are more opinionated on this.

**Review by wbamberg — APPROVED** [review]
👍 thank you @hamishwillee !

**Pull request merged**